code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,455 @@
+
+	var/mob/living/mob_occupant = occupant
+	if(mob_occupant)
+		// Eject dead people
+		if(mob_occupant.stat == DEAD)
+			open_machine()
+
+		if(!(world.time > despawn_world_time + 100))//+ 10 seconds
+			return
+
+		if(!mob_occupant.client && mob_occupant.stat < 2) //Occupant is living and has no client.
+			if(!control_computer)
+				find_control_computer(urgent = TRUE)//better hope you found it this time
+
+			despawn_occupant()
+
+// This function can not be undone; do not call this unless you are sure
+/obj/machinery/cryopod/proc/despawn_occupant()
+	var/mob/living/mob_occupant = occupant
+
+	if(istype(SSticker.mode, /datum/game_mode/cult))
+		if(GLOB.sac_mind == mob_occupant.mind)
+			var/list/possible_targets = list()
+			for(var/mob/living/carbon/human/H in GLOB.player_list)
+				if(H.mind && !is_convertable_to_cult(H) && !iscultist(H))
+					possible_targets += H.mind
+
+			possible_targets -= mob_occupant.mind
+			if(!possible_targets.len)
+				message_admins("Cult Sacrifice: Could not find unconvertable target, checking for convertable target.")
+				for(var/mob/living/carbon/human/player in GLOB.player_list)
+					if(player.mind && !iscultist(player))
+						possible_targets += player.mind
+
+			if(possible_targets.len > 0)
+				GLOB.sac_mind = pick(possible_targets)
+				if(!GLOB.sac_mind)
+					message_admins("Cult Sacrifice: ERROR -  Null target chosen!")
+				else
+					var/datum/job/sacjob = SSjob.GetJob(GLOB.sac_mind.assigned_role)
+					var/datum/preferences/sacface = GLOB.sac_mind.current.client.prefs
+					var/icon/reshape = get_flat_human_icon(null, sacjob, sacface)
+					reshape.Shift(SOUTH, 4)
+					reshape.Shift(EAST, 1)
+					reshape.Crop(7,4,26,31)
+					reshape.Crop(-5,-3,26,30)
+					GLOB.sac_image = reshape
+					for(var/datum/mind/H in SSticker.mode.cult)
+						if(H.current)
+							to_chat(H.current, "<span class='danger'>Nar'Sie</span> murmurs, <span class='cultlarge'>[occupant] is beyond your reach. Sacrifice [GLOB.sac_mind.current] instead...</span></span>")
+
+	//Update any existing objectives involving this mob.
+	for(var/datum/objective/O in GLOB.objectives)
+		// We don't want revs to get objectives that aren't for heads of staff. Letting
+		// them win or lose based on cryo is silly so we remove the objective.
+		if(istype(O,/datum/objective/mutiny) && O.target == mob_occupant.mind)
+			qdel(O)
+		else if(O.target && istype(O.target, /datum/mind))
+			if(O.target == mob_occupant.mind)
+				for(var/datum/mind/M in O.get_owners())
+					to_chat(M.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
+				O.target = null
+				spawn(10) //This should ideally fire after the occupant is deleted.
+					if(!O)
+						return
+					O.find_target()
+					O.update_explanation_text()
+					if(!(O.target))
+						O.owner.objectives -= O
+						qdel(O)
+					for(var/datum/mind/M in O.get_owners())
```

Comment:
You're using `O` right after potentially deleting it.

Reply:
This whole segment is kinda confusing (I hate nested ifs) but it's in the original code and I think it means "make a new target, if that person spawned bugs and isn't a target, delete from objectives, then announce all targets"
It should still work so long as O.target exists, right?